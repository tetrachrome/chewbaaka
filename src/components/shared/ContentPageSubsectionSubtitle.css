/**
 * ContentPageSubsectionSubtitle.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Jul 26, 2020
 */

h4.ContentPageSubsectionSubtitle {
  margin: 20px auto;
  line-height: 1.8;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 22px;
}

h4.ContentPageSubsectionSubtitle_DebugColor {
  background-color: goldenrod;
}
