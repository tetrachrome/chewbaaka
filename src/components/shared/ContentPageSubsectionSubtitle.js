/**
 * ContentPageSubsectionSubtitle.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Aug 07, 2020
 */

import React from 'react'

import { getElementStyleClassName } from '../../styling/styling'

import './ContentPageSubsectionSubtitle.css'

if ( process.env.NODE_ENV === 'development' )
  require('./ContentPageSubsectionSubtitle-debug.css')

export default function ContentPageSubsectionSubtitle({ children }) {
  return (
    <h4 className={getElementStyleClassName("ContentPageSubsectionSubtitle")}>
      {children}
    </h4>
  );
}
