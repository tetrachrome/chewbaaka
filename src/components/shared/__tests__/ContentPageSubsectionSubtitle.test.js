/**
 * ContentPageSubsectionSubtitle.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 26, 2020
 * Updated  : Jul 26, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageSubsectionSubtitle from '../ContentPageSubsectionSubtitle'

const titleText = "Hello, World";

test('renders ContentPageSubsectionSubtitle component', () => {
  render(
    <ContentPageSubsectionSubtitle>
      {titleText}
    </ContentPageSubsectionSubtitle>
  );

  const titleElement = screen.getByText(titleText);
  expect(titleElement).toBeInTheDocument();
});

test('ContentPageSubsectionSubtitle component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageSubsectionSubtitle>
        {titleText}
      </ContentPageSubsectionSubtitle>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
