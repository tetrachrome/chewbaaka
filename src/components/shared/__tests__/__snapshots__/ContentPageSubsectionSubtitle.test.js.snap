// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageSubsectionSubtitle component snapshot 1`] = `
<h4
  className="ContentPageSubsectionSubtitle ContentPageSubsectionSubtitle_DebugColor DebugWireframe"
>
  Hello, World
</h4>
`;
